layers/meta-balena-fsl-arm/recipes-support/hostapp-update-hooks/files/99-flash-bootloader-nitrogen8mm

```diff
@@ -0,0 +1,35 @@
+#!/bin/sh
+
+#
+# Script used by hostapps updater to flash bootloader onto internal media
+#
+
+set -o errexit
+
+# machine specific data
+uboot_file="imx-boot"
+uboot_block_size=1024
+uboot_seek_blocks=33
+device="/dev/mmcblk0boot0"
+
+update_files="uboot"
+
+for i in $update_files; do
+	current_update_file=$(eval echo \$${i}_file)
+	block_size=$(eval echo \$${i}_block_size)
+	seek_blocks=$(eval echo \$${i}_seek_blocks)
+
+	# calculate size and md5sum of the binary to update from the update bundle
+	update_size=$(ls -al /resin-boot/$current_update_file | awk '{print $5}')
+	update_md5sum=$(md5sum /resin-boot/$current_update_file | awk '{print $1'})
+
+	# calculate md5sum of the data already written to $device, using $update_size bytes and skipping $skip_bytes from $device
+	existing_md5sum=$(dd if=$device bs=1 skip=33792 count=$update_size status=none | md5sum | awk '{print $1}')
```

Comment:
Why hardcode `skip` here? Can it not be `skip_bytes=$((block_size * seek_blocks))` ?